src/command/root-command.ts

```diff
@@ -12,17 +14,31 @@ export class RootCommand {
   @ExternalOption('bee-api-url')
   public beeApiUrl!: string
 
+  @ExternalOption('config-folder')
+  public configFolder!: string
+
   // CLASS FIELDS
 
+  public appName = 'swarm-cli'
+
   public bee!: Bee
 
+  public commandConfig!: CommandConfig
+
   /**
    * Init Root command fields
    *
    * if BEE_API_URL environment variable has been set the CLI will use that connection string
    */
   protected init(): void {
-    this.beeApiUrl = this.beeApiUrl === defaultBeeApiUrl ? process.env.BEE_API_URL || this.beeApiUrl : this.beeApiUrl
+    this.commandConfig = new CommandConfig(this.appName, this.configFolder)
+    this.beeApiUrl = this.optionPassed(beeApiUrl)
+      ? this.beeApiUrl
+      : this.commandConfig.config.beeApiUrl || process.env.BEE_API_URL || this.beeApiUrl
     this.bee = new Bee(this.beeApiUrl)
   }
+
+  protected optionPassed(option: IOption): boolean {
+    return Boolean(process.argv.indexOf(`--${option.key}`) > -1)
```

Comment:
Consider using `includes` instead of `indexOf`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes